interview-questions.md

```diff
@@ -7,57 +7,81 @@ Then, research the question to expand on your answer. Even if you feel you have
+    $ rails generate migration add_foreign_key_to_models
+  As well as update my migration file and following the change with the terminal command:
+    $ db:migrate    
 
+  Researched answer: It is bad practice to make edits on the schema, which is not desiigned to be edited. Instead, if a change is to be made to a table in the schema, there is a command used on rails to permit the change. Once the command has been made, a migration file is generated that holds the desired changes to be made.
 
 
 4. Describe as many possible relationships between a model called Person and a model called Computer as you can think of. Based on each description, which model would hold the foreign key?
 
-  Your answer:
+  Your answer: In this scenario, I would need to determine which model has the 'has_many' method versus "belongs_to". A computer may have many users - people, so the person model will hold the foreign key.
+  Model - Computer:
+  1. Family_Computer      has many users (3 users)
+  2. Work_Computer        has many users (200 users)
+  3. Personal_Computer    has 1 user
+  Model - Person:
+  1. s.habteab.user - 1   Person.find(1) belongs to computer.find(1)
+  2. a.habteab.user - 1   Person.find(2) belongs to computer.find(1)
+  3. e.habteab.user - 1   Person.find(3) belongs to computer.find(1)
+  4. employee.user - 2    Person.find(4) belongs to computer.find(2)
+  204. employee.user - 2  Person.find(204) belongs to computer.find(2)
+  205. personal.user - 3  Person.find(205) belongs to computer.find(3)
+
+or you can think a person has many computers and in that case the computer has the foreign key.
+Model - Person:
+  1. Person1        has many computers (3)
+  2. Person2        has many computers (2)
+  3. Person3        has 1 computer
+  Model - Computer:
+  1. work_computer_A - 1   Computer.find(1) belongs to Person.find(1)
+  2. personal_computer - 1   Computer.find(2) belongs to Person.find(1)
+  3. work_computer_B - 1   Computer.find(3) belongs to Person.find(1)
+  4. personal_computer - 2    Computer.find(4) belongs to Person.find(2)
+  5. work_computer - 2  Computer.find(5) belongs to Person.find(2)
+  6. personal_computer - 3  Computer.find(6) belongs to Person.find(3)
 
   Researched answer:
-
+According to my researched answer, in the case that a computer has many users, the user would hold the foreign key. As noted, "A foreign key is a column in a relational database table that provides a link between data in two tables. It acts as a cross-reference between tables because it references the primary key of another table, thereby establishing a link between them. The foreign key always goes on the belongs_to side." and as a result, the foreign key lives in the person table in my specific example.scenario. 
 
 
 5. What is object-relational mapping?
 
-  Your answer:
+  Your answer: Object relational mapping (ORM) describes the relationships between models. 
```

Comment:
Almost! You definitely provided a better answer in your researched answer though! :)